iothub/service/src/Common/Amqp/ClientWebSocketTransport.cs

```diff
@@ -68,7 +68,41 @@ public override bool WriteAsync(TransportAsyncCallbackArgs args)
             return true;
         }
 
-        private async Task WriteAsyncCore(TransportAsyncCallbackArgs args)
+        public override bool ReadAsync(TransportAsyncCallbackArgs args)
+        {
+            ThrowIfNotOpen();
+
+            // Read with buffer list not supported
+            Fx.AssertAndThrow(args.Buffer != null, "must have buffer to read");
+            Fx.AssertAndThrow(args.CompletedCallback != null, "must have a valid callback");
+
+            Utils.ValidateBufferBounds(args.Buffer, args.Offset, args.Count);
+            args.Exception = null; // null out any exceptions
+
+            Task<int> taskResult = ReadImplAsync(args);
+            if (ReadTaskDone(taskResult, args))
+            {
+                return false;
+            }
+
+            taskResult.ToAsyncResult(s_onReadComplete, args);
+            return true;
+        }
+
+        public void Dispose()
+        {
+            if (!_isDisposed)
+            {
+                _webSocket.Abort();
```

Comment:
do you think it'll be an issue if we cann websocket.Abort() on an aborted socket? the doc only says that a websocket cannot be reused once aborted

Reply:
I don't know. I worry about changing it too much, though.